webapp/e2e/cypress/support/index.ts

```diff
@@ -1,9 +1,219 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import '@testing-library/cypress/add-commands';
+// import 'mattermost-webapp/e2e/cypress/support';
+
+// import '@testing-library/cypress/add-commands';
+
+// import 'cypress-wait-until';
+// import './ui_commands.ts';
+// import './api/index.ts';
 
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***********************************************************
+// Read more at: https://on.cypress.io/configuration
+// ***********************************************************
+
+/* eslint-disable no-loop-func */
+
+import '@testing-library/cypress/add-commands';
+import 'cypress-file-upload';
 import 'cypress-wait-until';
-import './server_api_commands.ts';
-import './ui_commands.ts';
-import './api/index.ts';
+import 'cypress-plugin-tab';
+import addContext from 'mochawesome/addContext';
+
+import 'mattermost-webapp/e2e/cypress/support/api';
+import 'mattermost-webapp/e2e/cypress/support/api_commands'; // soon to deprecate
+// import 'mattermost-webapp/e2e/cypress/support/client'; // have to keep this commented out until webpack/babel issue is fixed
```

Comment:
This is the part that needs to be commented out. Otherwise we could just do `import 'mattermost-webapp/e2e/cypress/support';`

```
Error: Webpack Compilation Error
./node_modules/mattermost-webapp/e2e/cypress/support/client.js 7:24
Module parse failed: Unexpected token (7:24)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| 
| class E2EClient extends Client4 {
>     doFetchWithResponse = async (url, options) => {
|         const {
|             body,
 @ ./cypress/support/index.ts 23:0-55
```

Reply:
I'm also wondering why the setup and spec files are in typescript in this repo, while they are in javascript in the webapp repo

Reply:
Is there a ticket to address this issue?

Reply:
typescript > javascript :wink:

Reply:
> Is there a ticket to address this issue?

@hanzei I've made one in Jira. Should it go in GitHub instead?
https://mattermost.atlassian.net/browse/MM-31410

Reply:
Jira is fine IMO. Would you mind linking that ticket in a comment?

Reply:
I moved the files to .ts from .js when I was copying stuff over.

Reply:
@hmhealey Did you have any trouble introducing the `E2EClient` ES6 class into the `e2e` folder? There is a webpack compile issue happening when the `mattermost-webapp/e2e/cypress/support/client.js` file is imported, shown in a comment above.

Typescript is currently being used as the `support/index.ts` entrypoint in this PR, but I also tried using a javascript file to import `client.js`, with no luck. The goal is to just have a one-liner import of `mattermost-webapp/e2e/cypress/support/index.js` in the `mattermost-plugin-demo/webapp/e2e/cypress/support/index.js` file, just as this is done with the regular webapp tests https://github.com/mattermost/mattermost-plugin-demo/blob/master/webapp/tests/setup.js .

@saturninoabril Do you have any thoughts on this? I'm not able to find any webpack settings to tweak to allow this, and I'm not sure how this is accomplished in the webapp repo.